go/store/nbs/store.go

```diff
@@ -1406,22 +1412,42 @@ func (nbs *NomsBlockStore) MarkAndSweepChunks(ctx context.Context, last hash.Has
 		return chunks.ErrNothingToCollect
 	}
 
-	specs, err := nbs.copyMarkedChunks(ctx, keepChunks)
-	if err != nil {
-		return err
-	}
-	if ctx.Err() != nil {
-		return ctx.Err()
+	destNBS := nbs
+	if dest != nil {
+		switch typed := dest.(type) {
+		case *NomsBlockStore:
+			destNBS = typed
+		case NBSMetricWrapper:
+			destNBS = typed.nbs
+		}
 	}
 
-	err = nbs.swapTables(ctx, specs)
+	specs, err := nbs.copyMarkedChunks(ctx, keepChunks, destNBS)
 	if err != nil {
 		return err
 	}
 	if ctx.Err() != nil {
 		return ctx.Err()
 	}
 
+	if destNBS == nbs {
```

Comment:
should add some clarifying comments here. The first half of the block is doing an old-style full-copy GC, the second half is moving chunks from the newGen to oldGen